packages/rest-hooks/src/resource/SimpleRecord.ts

```diff
@@ -0,0 +1,82 @@
+import { AbstractInstanceType } from '~/types';
+
+const DefinedMembersKey = Symbol('Defined Members');
+type Filter<T, U> = T extends U ? T : never;
+interface SimpleResourceMembers<T extends typeof SimpleRecord> {
+  [DefinedMembersKey]: Filter<keyof AbstractInstanceType<T>, string>[];
+}
+
+/** Immutable record that keeps track of which members are defined vs defaults. */
+export default abstract class SimpleRecord {
```

Comment:
this is all copy/pasted from SimpleResource, except the ABC check was moved out (since it used `pk` checking which isn't valid for this class).